api/meter_test.go

```diff
@@ -730,6 +731,148 @@ func TestAllAppsRelays(t *testing.T) {
 	}
 }
 
+func TestAppLatency(t *testing.T) {
+	todaysLatency := fakeTodaysLatency()
+	errBackendFailure := errors.New("backend error")
+
+	testCases := []struct {
+		name                string
+		requestedApp        string
+		todaysLatency       map[string][]Latency
+		expected            AppLatencyResponse
+		backendErr          error
+		expectedErr         error
+		expectedTodaysCalls int
+	}{
+		{
+			name:         "Correct latency data is returned",
+			requestedApp: "app1",
+			expected: AppLatencyResponse{
+				Application:  "app1",
+				From:         todaysLatency["app1"][0].Time,
+				To:           todaysLatency["app1"][len(todaysLatency["app1"])-1].Time,
+				DailyLatency: todaysLatency["app1"],
+			},
+		},
+		{
+			name:         "Returns error if requested app not found",
+			requestedApp: "app42",
+			expectedErr:  ErrAppLatencyNotFound,
+		},
+		{
+			name:         "Backend service error",
+			requestedApp: "app1",
+			backendErr:   errBackendFailure,
+			expectedErr:  ErrAppLatencyNotFound,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			fakeBackend := fakeBackend{
+				todaysLatency: todaysLatency,
+				err:           tc.backendErr,
+			}
+
+			relayMeter := NewRelayMeter(&fakeBackend, logger.New(), RelayMeterOptions{LoadInterval: 100 * time.Millisecond})
+			time.Sleep(200 * time.Millisecond)
+			got, err := relayMeter.AppLatency(tc.requestedApp)
+			if err != nil {
+				if tc.expectedErr == nil {
+					t.Fatalf("Unexpected error: %v", err)
+				}
+				if !strings.Contains(err.Error(), tc.expectedErr.Error()) {
```

Comment:
Not a blocker, but checking for specific errors is preferable to looking inside the error string.